VxWorks build failure in OS_FileChmod_Impl

**Describe the bug**
Failure occurs on a VxWorks 6.9 build:

```
/home/jphickey/cfecfs/github/osal/src/os/portable/os-impl-posix-files.c: In function 'OS_FileChmod_Impl':
/home/jphickey/cfecfs/github/osal/src/os/portable/os-impl-posix-files.c:202: error: too few arguments to function 'open'
make[7]: *** [osal/vxworks_impl/CMakeFiles/osal_vxworks_impl.dir/__/portable/os-impl-posix-files.c.obj] Error 1
```

Looks like this was recently introduced in the 2020-11-03 IC in via #639 ... 

**To Reproduce**
Build for MCP750 / vxworks 6.9 platform using standard method and default config.

**Expected behavior**
Build succeeds.

**Code snips**
Looks like VxWorks 6.9 doesn't implement `open()` per POSIX specs - it only has the 3 argument form (with mode) and not the 2 argument form.

**System observed on:**
VxWorks 6.9

**Additional context**
Hopefully we can get a CI that includes VxWorks....

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

--- a/src/os/portable/os-impl-posix-files.c
+++ b/src/os/portable/os-impl-posix-files.c
@@ -199,7 +199,7 @@ int32 OS_FileChmod_Impl(const char *local_path, uint32 access)
     int         fd;
 
     /* Open file to avoid filename race potential */
-    fd = open(local_path, O_RDONLY);
+    fd = open(local_path, O_RDONLY, 0);
     if (fd < 0)
     {
         return OS_ERROR;

